spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/IpIntegrationTests.java

```diff
@@ -164,6 +172,43 @@ public void testUdpInheritance() {
 		assertThat(udpMulticastOutboundChannelAdapterSpec2.get()).isInstanceOf(MulticastSendingMessageHandler.class);
 	}
 
+	@Test
+	public void testCloseStream() throws InterruptedException {
+		IntegrationFlow server = IntegrationFlows.from(Tcp.inboundGateway(Tcp.netServer(0)
+				.deserializer(new ByteArrayRawSerializer())))
+				.<byte[], String>transform(p -> "reply:" + new String(p).toUpperCase())
+				.bridge()
+				.get();
+		CountDownLatch latch = new CountDownLatch(1);
+		AtomicInteger port = new AtomicInteger();
+		class Listener implements ApplicationListener<TcpConnectionServerListeningEvent> {
+
+			@Override
+			public void onApplicationEvent(TcpConnectionServerListeningEvent event) {
+				port.set(event.getPort());
+				latch.countDown();
+			}
+
+		}
+		this.applicationContext.addApplicationListener(new Listener());
+		this.flowContext.registration(server)
+			.id("streamCloseServer")
+			.register();
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		IntegrationFlow client = IntegrationFlows.from(MessageChannels.direct())
+			.handle(Tcp.outboundGateway(Tcp.netClient("localhost", port.get())
+					.singleUseConnections(true)
+					.serializer(new ByteArrayRawSerializer()))
+				.closeStreamAfterSend(true))
+			.transform(Transformers.objectToString())
+			.get();
+		this.flowContext.registration(client)
+			.id("streamCloseClient")
+			.register();
+		MessagingTemplate messagingTemplate = new MessagingTemplate(client.getInputChannel());
```

Comment:
You can use `IntegrationFlowRegistration.getMessagingTemplate()` after the `.register()`